Document TOML modules and services (#2876)
esl/MongooseIM: doc/modules/mod_register.md

### Options

#### `modules.mod_register.iqdisc`
* **Syntax:** string
* **Default:** `"no_queue"`
* **Example:** `iqdisc = "one_queue"`

#### `modules.mod_register.access`
* **Syntax:** string
* **Default:** `"all"`
* **Example:** `access = "all"`

Defines which ACL should be used for checking if a chosen username is allowed for registration.

#### `modules.mod_register.welcome_message`
* **Syntax:** TOML table with the following keys: `"body"`, `"subject"` and string values.

> * **Default:** `[]`

Reviewer: What should be default value here? `[subject="", body=""]`?

> * **Example:** `welcome_message = ["Hello from MIM!", "Message body."]`

Reviewer 2: ```suggestion
* **Example:** `welcome_message = [subject = "Hello from MIM!", body = "Message body."]`
```

Body and subject of a `<message>` stanza sent to new users. Only one of the fields (but non-empty) is mandatory for the message to be sent.

#### `modules.mod_register.registration_watchers`
* **Syntax:** array of strings

> * **Default:** `[]`

Reviewer: What should be default value here?

* **Example:** `registration_watchers = ["JID1", "JID2"]`

List of JIDs, which should receive a `<message>` notification about every successful registration.

#### `modules.mod_register.password_strength`
* **Syntax:** non-negative integer
* **Default:** `0`
* **Example:** `password_strength = 32`

Specifies minimal entropy of allowed password.
Entropy is measured with `ejabberd_auth:entropy/1`.
Recommended minimum is 32.
The entropy calculation algorithm is described in a section below.

#### `modules.mod_register.ip_access`

> * **Syntax:** Array of TOML tables with the following keys: `address`, `policy` and following values: {address = `string`, policy = `values: "allow", "deny"`}.

Reviewer: I stuck just to `string` here as the validator calls `inet:parse_address/1` what can take empty string, however throws `{error,einval}`.

Reviewer 3: ```suggestion
* **Syntax:** Array of TOML tables with the following mandatory content:
  - `address` - string, IP address
  - `policy` - string, one of: `"allow"`, `"deny"`.
```

Please check how it looks before committing the suggestion.

> * **Default:** `[]`

Reviewer: What should be default value here?

Reviewer 3: Empty list, it's correct.

* **Example:** `ip_access = [{address = "127.0.0.0/8", policy = "allow"}, {address = "0.0.0.0/0", policy = "deny"}]`

Access list for specified IPs or networks. 
Default value allows registration from every IP.

### Example configuration

Allow registrations from localhost:
``` 
[modules.mod_register]

>   welcome_message = ["", ""]

Reviewer 2: same as before, `subject` and `body` key are required

  ip_access = [
    {address = "127.0.0.1", policy = "allow"}
  ]
  access = "register"
```

Deny registration from network 10.20.0.0 with mask 255.255.0.0.
```
[modules.mod_register]

>   welcome_message = ["", ""]

Reviewer 2: same as above

#### `modules.mod_register.welcome_message`
* **Syntax:** TOML table with the following keys: `"body"`, `"subject"` and string values.
* **Default:** `[]`
* **Example:** `welcome_message = [subject = "Hello from MIM!", body = "Message body."]`

Body and subject of a `<message>` stanza sent to new users. Only one of the fields (but non-empty) is mandatory for the message to be sent.

#### `modules.mod_register.registration_watchers`
* **Syntax:** array of strings
* **Default:** `[]`
* **Example:** `registration_watchers = ["JID1", "JID2"]`

List of JIDs, which should receive a `<message>` notification about every successful registration.

#### `modules.mod_register.password_strength`
* **Syntax:** non-negative integer
* **Default:** `0`
* **Example:** `password_strength = 32`

Specifies minimal entropy of allowed password.
Entropy is measured with `ejabberd_auth:entropy/1`.
Recommended minimum is 32.
The entropy calculation algorithm is described in a section below.

#### `modules.mod_register.ip_access`
* **Syntax:** Array of TOML tables with the following keys: `address`, `policy` and following values: {address = `string`, policy = `values: "allow", "deny"`}.
* **Default:** `[]`

> * **Example:** `ip_access = [{address = "127.0.0.0/8", policy = "allow"}, {address = "0.0.0.0/0", policy = "deny"}]`

Reviewer 3: Maybe split in two lines? Just a suggestion. arrays can be multi-line.

Access list for specified IPs or networks. 
Default value allows registration from every IP.

### Example configuration

Allow registrations from localhost:
``` 
[modules.mod_register]

>   welcome_message = [subject = "", body = ""]

Reviewer 3: ```suggestion
  welcome_message = {subject = "", body = ""}
```

but this is strange as it's the same as default = no effect, any chance of another example? ;)

Reviewer: applying more meaningful example

  ip_access = [
    {address = "127.0.0.1", policy = "allow"}
  ]
  access = "register"
```

Deny registration from network 10.20.0.0 with mask 255.255.0.0.
```
[modules.mod_register]

>   welcome_message = [subject = "", body = ""]

Reviewer 3: ```suggestion
  welcome_message = {subject = "", body = ""}
```
maybe remove this from the second example (or both)?

Reviewer: removing
